nightingale_navigation/src/room_runner.py

```diff
@@ -0,0 +1,288 @@
+        self.client.cancel_all_goals()
+        result = RoomRunnerResult()
+        result.error_code = self.ErrorCode.PREEMPTED_BY_CALLER.value
+        self.server.set_preempted(result)
+
+    def generate_goal_info(self, pose, phase):
+        msg = MoveBaseGoal()
+        msg.target_pose = pose
+
+        def fb_cb(fb):
+            target = np.array([pose.pose.position.x, pose.pose.position.y])
+            cur = np.array(
+                [fb.base_position.pose.position.x, fb.base_position.pose.position.y]
+            )
+            euclidean_distance = np.linalg.norm(target - cur)
+            ret = RoomRunnerFeedback()
+            ret.euclidean_distance_to_goal = euclidean_distance
+            ret.current_phase = phase
+            self.server.publish_feedback(ret)
+
+        a_cb = lambda: rospy.loginfo(
+            "Move Base activated the action", logger_name=self.logger_name
+        )
+        return msg, fb_cb, a_cb
+
+    def publish_initial_pose(self, x, y, yaw):
+        msg = PoseWithCovarianceStamped()
+        msg.header.stamp = rospy.Time.now()
+        msg.header.frame_id = "map"
+        msg.pose.pose.position.x = x
+        msg.pose.pose.position.y = y
+        quat = tf_conversions.transformations.quaternion_from_euler(0, 0, yaw)
+        msg.pose.pose.orientation.x = quat[0]
+        msg.pose.pose.orientation.y = quat[1]
+        msg.pose.pose.orientation.z = quat[2]
+        msg.pose.pose.orientation.w = quat[3]
+        msg.pose.covariance = self.INITAL_COV.copy()
+
+        pub = rospy.Publisher("/initialpose", PoseWithCovarianceStamped, queue_size=10)
+        # Fast rate for checking if amcl is up
+        rate = rospy.Rate(100)
+        while pub.get_num_connections() == 0:
+            rospy.loginfo_throttle(5, "Waiting for amcl to subscribe to /initialpose")
+            rate.sleep()
+        pub.publish(msg)
+
+    def call_door_lookup(self, name):
+        # TODO Service Call, currently, just always returns the same pose
+        ret = PoseStamped()
+        ret.header.stamp = rospy.Time.now()
+        ret.header.frame_id = "map"
+        ret.pose.position.x = 4.80
+        ret.pose.position.y = 12.24
+        ret.pose.position.z = 0.0
+        ret.pose.orientation.x = 0.0
+        ret.pose.orientation.y = 0.0
+        ret.pose.orientation.z = 0.0
+        ret.pose.orientation.w = 1.0
+        return ret
+
+    def call_bedside_lookup(self, name):
+        # TODO Service Call, currently, just always returns the same pose
+        ret = PoseStamped()
+        ret.header.stamp = rospy.Time.now()
+        ret.header.frame_id = "map"
+        ret.pose.position.x = 4.80
+        ret.pose.position.y = -3.34
+        ret.pose.position.z = 0.0
+        ret.pose.orientation.x = 0.0
+        ret.pose.orientation.y = 0.0
+        ret.pose.orientation.z = 0.0
+        ret.pose.orientation.w = 1.0
+        return ret
+
+    def read_pose(self):
```

Comment:
Would it be useful to have some parameter to change the pose filename name? I'm wondering if it might be annoying to have to go through a file which contains all of the poses from every run and splitting up the files for the runs might make things more organized

Reply:
After reading the code a bit more it seems like the pose is always overwritten so that there is only a single line in the file at a time?

Reply:
second comment is correct. also the filename is a param specified in the launch file